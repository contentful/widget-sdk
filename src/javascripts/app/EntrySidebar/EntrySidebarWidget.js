import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { css } from 'emotion';
import tokens from '@contentful/forma-36-tokens';
import classNames from 'classnames';

const headerClassName = css({
  fontSize: tokens.spacingS,
  display: 'flex',
  justifyContent: 'space-between',
  borderBottom: `1px solid ${tokens.colorElementDark}`,
  marginBottom: tokens.spacingL,
  marginTop: tokens.spacingL,
  alignItems: 'center',
});

const titleClassName = css({
  fontSize: tokens.fontSizeS,
  fontWeight: tokens.fontWeightMedium,
  textTransform: 'uppercase',
  color: tokens.colorTextLight,
  borderBottom: 0,
  margin: 0,
  lineHeight: 2,
  letterSpacing: '1px',
});

export default class EntrySidebarWidget extends Component {
  static propTypes = {
    title: PropTypes.string.isRequired,
    testId: PropTypes.string,
    headerNode: PropTypes.node,
    children: PropTypes.node,
  };

  static defaultProps = {
    testId: 'cf-entity-sidebar',
  };

  render() {
    const { headerNode, testId, title, children, ...restProps } = this.props;
    return (
      <div data-test-id={testId} {...restProps}>
        <header className={classNames(headerClassName, 'entity-sidebar__header')}>
          <h2 className={classNames(titleClassName, 'entity-sidebar__heading')}>{title}</h2>
          {headerNode}
        </header>
        {children}
      </div>
    );
  }
}
