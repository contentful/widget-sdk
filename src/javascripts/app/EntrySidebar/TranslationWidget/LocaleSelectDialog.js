/* eslint "rulesdir/restrict-inline-styles": "warn" */
import React from 'react';
import PropTypes from 'prop-types';
import _ from 'lodash';
import { Modal, CheckboxField, Button, TextLink } from '@contentful/forma-36-react-components';

class LocaleSelectDialog extends React.Component {
  static propTypes = {
    initialLocales: PropTypes.array,
    onClose: PropTypes.func,
    onUpdate: PropTypes.func,
  };

  state = {
    locales: this.props.initialLocales,
    selectAllToggle:
      this.props.initialLocales.filter((locale) => locale.active).length >=
      this.props.initialLocales.length,
  };

  handleConfirm = () => {
    this.props.onUpdate(this.state.locales);
    this.props.onClose();
  };

  handleSelectAllToggle = () => {
    const active = !this.state.selectAllToggle;

    this.setState({
      selectAllToggle: active,
      locales: [
        ...this.state.locales.map((locale) => (locale.default ? locale : { ...locale, active })),
      ],
    });
  };

  handleChangeLocale(locale) {
    this.setState({
      locales: [
        ...this.state.locales.map((initialLocale) =>
          initialLocale.code === locale.code
            ? {
                ...locale,
                active: !initialLocale.active,
              }
            : initialLocale
        ),
      ],
    });
  }

  renderSelectAllToggle() {
    const toggleLabel = this.state.selectAllToggle ? 'Deselect all' : 'Select all';

    return <TextLink onClick={() => this.handleSelectAllToggle()}>{toggleLabel}</TextLink>;
  }

  renderLocaleField(locale) {
    return (
      <li key={locale.code}>
        <CheckboxField
          labelText={`${locale.name} (${locale.code})`}
          disabled={locale.default}
          name={locale.code}
          value={locale.code}
          checked={locale.active}
          onChange={() => this.handleChangeLocale(locale)}
          id={locale.code}
          testId="locale"
        />
      </li>
    );
  }

  render() {
    return (
      <React.Fragment>
        <Modal.Header title="Translation" />
        <Modal.Content>
          {this.state.locales.length > 1 && (
            <div className="f36-margin-bottom--m" style={{ display: 'flex' }}>
              <span className="f36-font-weight--demi-bold" style={{ flex: '1 0 0' }}>
                Select locale(s)
              </span>
              {this.renderSelectAllToggle()}
            </div>
          )}
          <ul>
            {_.orderBy(this.state.locales, ['default', 'name'], ['desc', 'asc']).map((locale) =>
              this.renderLocaleField(locale)
            )}
          </ul>
        </Modal.Content>
        <Modal.Controls>
          <Button buttonType="positive" onClick={this.handleConfirm} testId="save-cta">
            Save
          </Button>
          <Button buttonType="muted" onClick={this.props.onClose} testId="cancel-cta">
            Cancel
          </Button>
        </Modal.Controls>
      </React.Fragment>
    );
  }
}

const LocaleSelectDialogModal = (props) => (
  <Modal isShown={props.isShown} onClose={props.onClose} testId="locale-select-dialog">
    {() => (
      <LocaleSelectDialog
        onClose={props.onClose}
        onUpdate={props.onUpdate}
        initialLocales={props.initialLocales}
      />
    )}
  </Modal>
);

LocaleSelectDialogModal.propTypes = {
  initialLocales: PropTypes.array,
  onClose: PropTypes.func,
  onUpdate: PropTypes.func,
  isShown: PropTypes.bool,
};

export default LocaleSelectDialogModal;
