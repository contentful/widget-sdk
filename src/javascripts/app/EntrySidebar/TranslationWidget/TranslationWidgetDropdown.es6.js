import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { orderBy } from 'lodash';
import { Select, Option } from '@contentful/forma-36-react-components';
import SidebarEventTypes from 'app/EntrySidebar/SidebarEventTypes.es6';
import { truncate } from 'utils/StringUtils.es6';
import { track } from 'analytics/Analytics.es6';

const sortLocales = locales => orderBy(locales, ['default', 'name'], ['desc', 'asc']);

export default class TranslationDropdownWidget extends Component {
  static propTypes = {
    emitter: PropTypes.shape({
      emit: PropTypes.func.isRequired
    }).isRequired,
    localeData: PropTypes.shape({
      privateLocales: PropTypes.arrayOf(
        PropTypes.shape({
          name: PropTypes.string.isRequired,
          code: PropTypes.string.isRequired,
          internal_code: PropTypes.string.isRequired,
          default: PropTypes.bool.isRequired
        }).isRequired
      ).isRequired,
      errors: PropTypes.object.isRequired,
      focusedLocale: PropTypes.shape({
        internal_code: PropTypes.string.isRequired
      }).isRequired
    }).isRequired
  };

  numErrors = code => {
    if (!this.props.localeData.errors[code]) {
      return 0;
    }
    return this.props.localeData.errors[code].length;
  };

  localeName = ({ internal_code: internalCode, code, name }) => {
    const numErrors = this.numErrors(internalCode);
    if (numErrors > 0) {
      const errors = numErrors > 1 ? 'errors' : 'error';
      return `${truncate(name, 20)} (${code}) - ${numErrors} ${errors}`;
    } else {
      return `${name} (${code})`;
    }
  };

  handleChange = event => {
    const focusedLocaleCode = event.target.value;
    const newLocale = this.props.localeData.privateLocales.find(locale => {
      return locale.internal_code === focusedLocaleCode;
    });
    this.props.emitter.emit(SidebarEventTypes.UPDATED_FOCUSED_LOCALE, newLocale);
    track('translation_sidebar:change_focused_locale', { currentMode: 'single' });
  };

  render() {
    const {
      privateLocales,
      focusedLocale: { internal_code: focusedLocaleCode }
    } = this.props.localeData;
    return (
      <div>
        <Select id="optionSelect" onChange={this.handleChange} value={focusedLocaleCode}>
          {sortLocales(privateLocales).map(locale => (
            <Option key={locale.internal_code} value={locale.internal_code}>
              {this.localeName(locale)}
            </Option>
          ))}
        </Select>
      </div>
    );
  }
}
