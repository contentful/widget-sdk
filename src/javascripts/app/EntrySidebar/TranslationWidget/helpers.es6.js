import { orderBy } from 'lodash';

export const orderLocales = locales =>
  orderBy(locales, ['default', 'internal_code'], ['desc', 'asc']);
