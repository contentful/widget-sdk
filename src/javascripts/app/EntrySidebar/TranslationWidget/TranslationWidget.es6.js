import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { orderBy } from 'lodash';
import EntrySidebarWidget from '../EntrySidebarWidget.es6';
import { Select, Option } from '@contentful/forma-36-react-components';
import { getModule } from 'NgRegistry.es6';
import SidebarEventTypes from 'app/EntrySidebar/SidebarEventTypes.es6';

const TheLocaleStore = getModule('TheLocaleStore');

export default class TranslationSidebarWidget extends Component {
  static propTypes = {
    emitter: PropTypes.shape({
      emit: PropTypes.func.isRequired
    }).isRequired,
    localeErrors: PropTypes.object
  };

  constructor(props) {
    super(props);
    this.locales = TheLocaleStore.getLocales();
  }

  numErrors = code => {
    if (!this.props.localeErrors || !this.props.localeErrors[code]) {
      return 0;
    }
    return this.props.localeErrors[code].length;
  };

  localeName = ({ internal_code: internalCode, code, name }) => {
    const baseName = `${name} (${code})`;
    const numErrors = this.numErrors(internalCode);
    return numErrors > 0
      ? `${baseName} (${numErrors} ${numErrors > 1 ? 'errors' : 'error'})`
      : baseName;
  };

  handleChange = event => {
    const newLocale = event.target.value;
    TheLocaleStore.setCurrentLocale(newLocale);
    this.props.emitter.emit(SidebarEventTypes.UPDATED_CURRENT_LOCALE, newLocale);
  };

  componentWillUnmount = () => {
    TheLocaleStore.resetCurrentLocale();
  };

  render() {
    return (
      <EntrySidebarWidget testId="sidebar-translation-widget" title="Translation">
        <Select id="optionSelect" onChange={this.handleChange}>
          {orderBy(this.locales, ['default', 'code'], ['desc', 'asc']).map(locale => (
            <Option key={locale.code} value={locale.code}>
              {this.localeName(locale)}
            </Option>
          ))}
        </Select>
      </EntrySidebarWidget>
    );
  }
}
