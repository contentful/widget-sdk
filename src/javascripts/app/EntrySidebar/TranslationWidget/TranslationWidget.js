import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { css } from 'emotion';
import tokens from '@contentful/forma-36-tokens';

import TranslationWidgetPills from './TranslationWidgetPills';
import EntrySidebarWidget from '../EntrySidebarWidget';
import SidebarEventTypes from 'app/EntrySidebar/SidebarEventTypes';
import TranslationWidgetDropdown from './TranslationWidgetDropdown';
import { Select, Option } from '@contentful/forma-36-react-components';
import { track } from 'analytics/Analytics';

const Tab = {
  MULTI: 'multiple',
  SINGLE: 'single'
};

const selectStyles = css({
  '& select': {
    backgroundColor: 'transparent',
    textAlignLast: 'right',
    border: '0',
    padding: `0 ${tokens.spacingL} 0 0`,
    height: tokens.spacingL
  },
  '& select:focus': {
    border: 0,
    boxShadow: 'none'
  },
  '& svg': {
    right: 0
  }
});

export default class TranslationSidebarWidget extends Component {
  static propTypes = {
    localeData: PropTypes.shape({
      isSingleLocaleModeOn: PropTypes.bool.isRequired
    }).isRequired,
    emitter: PropTypes.shape({
      emit: PropTypes.func.isRequired
    }).isRequired
  };

  onSelectChange = event => {
    const {
      target: { value }
    } = event;
    this.props.emitter.emit(SidebarEventTypes.SET_SINGLE_LOCALE_MODE, value === Tab.SINGLE);
    track('translation_sidebar:toggle_widget_mode', { currentMode: value });
  };

  headerNode = () => (
    <Select
      value={this.props.localeData.isSingleLocaleModeOn ? Tab.SINGLE : Tab.MULTI}
      onChange={this.onSelectChange}
      width="auto"
      className={selectStyles}>
      <Option value={Tab.MULTI}>Multiple locales</Option>
      <Option value={Tab.SINGLE}>Single locale</Option>
    </Select>
  );

  render() {
    const { isSingleLocaleModeOn } = this.props.localeData;
    const CurrentTranslationWidget = isSingleLocaleModeOn
      ? TranslationWidgetDropdown
      : TranslationWidgetPills;

    return (
      <EntrySidebarWidget
        testId="sidebar-translation-widget"
        title="Translation"
        headerNode={this.headerNode()}>
        <CurrentTranslationWidget {...this.props} />
      </EntrySidebarWidget>
    );
  }
}
