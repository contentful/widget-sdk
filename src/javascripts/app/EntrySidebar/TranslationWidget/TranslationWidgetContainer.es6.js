import React from 'react';
import _ from 'lodash';
import TranslationWidget from './TranslationWidget.es6';
import ModalLauncher from 'app/common/ModalLauncher.es6';
import LocaleSelectDialog from './LocaleSelectDialog.es6';
import { PropTypes } from 'prop-types';
import { track } from 'analytics/Analytics.es6';
import { getModule } from 'NgRegistry.es6';

const TheLocaleStore = getModule('TheLocaleStore');

/**
 * Returns an array of copies of `locales` with an additional
 * `active` property.
 */
function getLocalesWithActiveFlag(locales) {
  return _.map(locales, function(locale) {
    return _.extend(
      {
        active: TheLocaleStore.isLocaleActive(locale)
      },
      locale
    );
  });
}

export default class TranslationWidgetContainer extends React.Component {
  static propTypes = {
    emitter: PropTypes.shape({
      emit: PropTypes.func.isRequired
    }).isRequired,
    localeErrors: PropTypes.object
  };

  constructor(props) {
    super(props);
    this.state = {
      locales: []
    };
  }

  componentDidMount() {
    this.updateLocales();
  }

  updateLocales = () => {
    this.setState({
      locales: TheLocaleStore.getActiveLocales()
    });
  };

  onChange = async () => {
    const locales = getLocalesWithActiveFlag(TheLocaleStore.getPrivateLocales());

    const onUpdate = locales => {
      const activeLocales = locales.filter(locale => locale.active);
      TheLocaleStore.setActiveLocales(activeLocales);
      this.updateLocales();
      track('translation_sidebar:update_active_locales', { currentMode: 'multiple' });
    };

    await ModalLauncher.open(({ onClose, isShown }) => (
      <LocaleSelectDialog
        initialLocales={locales}
        isShown={isShown}
        onClose={onClose}
        onUpdate={onUpdate}
      />
    ));
  };

  onLocaleDeactivation = locale => {
    TheLocaleStore.deactivateLocale(locale);
    this.updateLocales();
    track('translation_sidebar:deselect_active_locale', { currentMode: 'multiple' });
  };

  render() {
    return (
      <TranslationWidget
        locales={this.state.locales}
        onChange={this.onChange}
        onLocaleDeactivation={this.onLocaleDeactivation}
        emitter={this.props.emitter}
        localeErrors={this.props.localeErrors}
      />
    );
  }
}
