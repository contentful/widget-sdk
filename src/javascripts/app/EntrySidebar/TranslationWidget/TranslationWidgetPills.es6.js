import React, { Component } from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import SidebarEventTypes from 'app/EntrySidebar/SidebarEventTypes.es6';
import ModalLauncher from 'app/common/ModalLauncher.es6';
import LocaleSelectDialog from './LocaleSelectDialog.es6';
import { Pill, TextLink } from '@contentful/forma-36-react-components';
import { track } from 'analytics/Analytics.es6';
import { orderBy } from 'lodash';

const sortLocales = locales => orderBy(locales, ['default', 'code'], ['desc', 'asc']);

const trackTranslationSidebarEvent = (action, data) =>
  track(`translation_sidebar:${action}`, {
    currentMode: 'multiple',
    ...data
  });

const localesPropType = PropTypes.arrayOf(
  PropTypes.shape({
    code: PropTypes.string.isRequired,
    default: PropTypes.bool.isRequired
  }).isRequired
).isRequired;

export default class TranslationWidgetPills extends Component {
  static propTypes = {
    emitter: PropTypes.shape({
      emit: PropTypes.func.isRequired
    }).isRequired,
    localeData: PropTypes.shape({
      activeLocales: localesPropType,
      privateLocales: localesPropType,
      isLocaleActive: PropTypes.func.isRequired
    })
  };

  onLocaleDeactivation = locale => {
    const activeLocaleCount = this.props.localeData.activeLocales.length;
    trackTranslationSidebarEvent('deselect_active_locale', {
      previousActiveLocaleCount: activeLocaleCount,
      currentActiveLocaleCount: activeLocaleCount - 1
    });
    this.props.emitter.emit(SidebarEventTypes.DEACTIVATED_LOCALE, locale);
  };

  onChange = async () => {
    const { privateLocales, isLocaleActive } = this.props.localeData;

    const initialLocales = privateLocales.map(locale => ({
      ...locale,
      active: isLocaleActive(locale)
    }));

    const onUpdate = locales => {
      const currentActiveLocales = locales.filter(l => l.active);
      trackTranslationSidebarEvent('update_active_locales', {
        previousActiveLocaleCount: this.props.localeData.activeLocales.length,
        currentActiveLocaleCount: currentActiveLocales.length
      });
      this.props.emitter.emit(SidebarEventTypes.SET_ACTIVE_LOCALES, currentActiveLocales);
    };

    await ModalLauncher.open(({ onClose, isShown }) => (
      <LocaleSelectDialog
        initialLocales={initialLocales}
        isShown={isShown}
        onClose={onClose}
        onUpdate={onUpdate}
      />
    ));
  };

  render() {
    return (
      <div className="pill-list entity-sidebar__translation-pills">
        {sortLocales(this.props.localeData.activeLocales).map(locale => (
          <div
            key={locale.code}
            className={classNames('entity-sidebar__translation-pill', {
              'x--default': locale.default
            })}>
            <Pill
              testId="deactivate-translation"
              status="default"
              label={locale.code}
              onClose={locale.default ? undefined : () => this.onLocaleDeactivation(locale)}
            />
          </div>
        ))}
        <TextLink testId="change-translation" onClick={this.onChange}>
          Change
        </TextLink>
      </div>
    );
  }
}
