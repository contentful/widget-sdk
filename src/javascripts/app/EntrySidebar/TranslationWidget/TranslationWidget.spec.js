import React from 'react';
import Enzyme from 'enzyme';
import TranslationWidget from './TranslationWidget.es6';
import EntrySidebarWidget from '../EntrySidebarWidget.es6';
import SidebarEventTypes from 'app/EntrySidebar/SidebarEventTypes.es6';
import { track } from 'analytics/Analytics.es6';

describe('EntrySidebar/TranslationWidget', () => {
  const locales = [
    { internal_code: 'en-US', code: 'en', name: 'English (United States)', default: true },
    { internal_code: 'de-DE', code: 'de', name: 'German (Germany)', default: false },
    { internal_code: 'es-AR', code: 'es', name: 'Spanish (Argentina)', default: false },
    { internal_code: 'ru', code: 'ru', name: 'Russian', default: false }
  ];

  const props = {
    localeData: {
      activeLocales: locales,
      privateLocales: locales,
      errors: {},
      focusedLocale: locales[0],
      isLocaleActive: () => {},
      isSingleLocaleModeOn: false
    },
    emitter: {
      emit: jest.fn()
    }
  };
  const render = () => Enzyme.shallow(<TranslationWidget {...props} />);

  describe('when single locale mode is on', () => {
    beforeEach(() => {
      props.localeData.isSingleLocaleModeOn = true;
    });

    it('should match snapshot', () => {
      expect(render()).toMatchSnapshot();
    });

    describe('and the locale mode is change', () => {
      beforeEach(() => {
        const headerNode = Enzyme.shallow(
          render()
            .find(EntrySidebarWidget)
            .prop('headerNode')
        );
        headerNode.find('select').prop('onChange')({ target: { value: 'multiple' } });
      });

      it('emits the SET_SINGLE_LOCALE_MODE event with the new value', () => {
        expect(props.emitter.emit).toHaveBeenCalledWith(
          SidebarEventTypes.SET_SINGLE_LOCALE_MODE,
          false
        );
      });

      it('tracks the change event', () => {
        expect(track).toHaveBeenCalledWith('translation_sidebar:toggle_widget_mode', {
          currentMode: 'multiple'
        });
      });
    });
  });

  describe('when single locale mode is off', () => {
    beforeEach(() => {
      props.localeData.isSingleLocaleModeOn = false;
    });

    it('should match snapshot', () => {
      expect(render()).toMatchSnapshot();
    });

    describe('and the locale mode is changed', () => {
      beforeEach(() => {
        const headerNode = Enzyme.shallow(
          render()
            .find(EntrySidebarWidget)
            .prop('headerNode')
        );
        headerNode.find('select').prop('onChange')({ target: { value: 'single' } });
      });

      it('emits the SET_SINGLE_LOCALE_MODE event with the new value', () => {
        expect(props.emitter.emit).toHaveBeenCalledWith(
          SidebarEventTypes.SET_SINGLE_LOCALE_MODE,
          true
        );
      });

      it('tracks the change event', () => {
        expect(track).toHaveBeenCalledWith('translation_sidebar:toggle_widget_mode', {
          currentMode: 'single'
        });
      });
    });
  });
});
