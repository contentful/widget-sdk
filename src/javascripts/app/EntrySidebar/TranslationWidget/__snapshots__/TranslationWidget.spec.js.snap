// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EntrySidebar/TranslationWidget when single locale mode is off should match snapshot 1`] = `
<EntrySidebarWidget
  headerNode={
    <Select
      className="css-1frdblk-selectStyles"
      hasError={false}
      isDisabled={false}
      onChange={[Function]}
      required={false}
      testId="cf-ui-select"
      value="multiple"
      width="auto"
      willBlurOnEsc={true}
    >
      <Option
        testId="cf-ui-select-option"
        value="multiple"
      >
        Multiple locales
      </Option>
      <Option
        testId="cf-ui-select-option"
        value="single"
      >
        Single locale
      </Option>
    </Select>
  }
  testId="sidebar-translation-widget"
  title="Translation"
>
  <TranslationWidgetPills
    emitter={
      Object {
        "emit": [MockFunction],
      }
    }
    localeData={
      Object {
        "activeLocales": Array [
          Object {
            "code": "en",
            "default": true,
            "internal_code": "en-US",
            "name": "English (United States)",
          },
          Object {
            "code": "de",
            "default": false,
            "internal_code": "de-DE",
            "name": "German (Germany)",
          },
          Object {
            "code": "es",
            "default": false,
            "internal_code": "es-AR",
            "name": "Spanish (Argentina)",
          },
          Object {
            "code": "ru",
            "default": false,
            "internal_code": "ru",
            "name": "Russian",
          },
        ],
        "errors": Object {},
        "focusedLocale": Object {
          "code": "en",
          "default": true,
          "internal_code": "en-US",
          "name": "English (United States)",
        },
        "isLocaleActive": [Function],
        "isSingleLocaleModeOn": false,
        "privateLocales": Array [
          Object {
            "code": "en",
            "default": true,
            "internal_code": "en-US",
            "name": "English (United States)",
          },
          Object {
            "code": "de",
            "default": false,
            "internal_code": "de-DE",
            "name": "German (Germany)",
          },
          Object {
            "code": "es",
            "default": false,
            "internal_code": "es-AR",
            "name": "Spanish (Argentina)",
          },
          Object {
            "code": "ru",
            "default": false,
            "internal_code": "ru",
            "name": "Russian",
          },
        ],
      }
    }
  />
</EntrySidebarWidget>
`;

exports[`EntrySidebar/TranslationWidget when single locale mode is on should match snapshot 1`] = `
<EntrySidebarWidget
  headerNode={
    <Select
      className="css-1frdblk-selectStyles"
      hasError={false}
      isDisabled={false}
      onChange={[Function]}
      required={false}
      testId="cf-ui-select"
      value="single"
      width="auto"
      willBlurOnEsc={true}
    >
      <Option
        testId="cf-ui-select-option"
        value="multiple"
      >
        Multiple locales
      </Option>
      <Option
        testId="cf-ui-select-option"
        value="single"
      >
        Single locale
      </Option>
    </Select>
  }
  testId="sidebar-translation-widget"
  title="Translation"
>
  <TranslationDropdownWidget
    emitter={
      Object {
        "emit": [MockFunction],
      }
    }
    localeData={
      Object {
        "activeLocales": Array [
          Object {
            "code": "en",
            "default": true,
            "internal_code": "en-US",
            "name": "English (United States)",
          },
          Object {
            "code": "de",
            "default": false,
            "internal_code": "de-DE",
            "name": "German (Germany)",
          },
          Object {
            "code": "es",
            "default": false,
            "internal_code": "es-AR",
            "name": "Spanish (Argentina)",
          },
          Object {
            "code": "ru",
            "default": false,
            "internal_code": "ru",
            "name": "Russian",
          },
        ],
        "errors": Object {},
        "focusedLocale": Object {
          "code": "en",
          "default": true,
          "internal_code": "en-US",
          "name": "English (United States)",
        },
        "isLocaleActive": [Function],
        "isSingleLocaleModeOn": true,
        "privateLocales": Array [
          Object {
            "code": "en",
            "default": true,
            "internal_code": "en-US",
            "name": "English (United States)",
          },
          Object {
            "code": "de",
            "default": false,
            "internal_code": "de-DE",
            "name": "German (Germany)",
          },
          Object {
            "code": "es",
            "default": false,
            "internal_code": "es-AR",
            "name": "Spanish (Argentina)",
          },
          Object {
            "code": "ru",
            "default": false,
            "internal_code": "ru",
            "name": "Russian",
          },
        ],
      }
    }
  />
</EntrySidebarWidget>
`;
